lesson2/README.md

```diff
@@ -1,14 +1,102 @@
+2. **XML Layout files** are where you define the frontend layout of your screens.
+3. **Configuration files** are used by Android Studio to configure your app.
+
+You'll be writing most of your code in Java files.
+### 1. Java Files
+When you interact with an app on your phone, there are different "screens" that you can go to. Snapchat, for example has a "friends list" screen, a "picture-taking" screen, a "settings" screen, and so on.  The settings screen displays different things and needs completely different logic than the picture-taking screen.  Android separates these "screens" into different files called **activities** and **fragments**.
+
+#### [Activities](https://developer.android.com/guide/components/activities.html) (ex. MainActivity.java)
+
+ An activity provides a screen that users can interact with in order to do something, such as dial the phone, take a photo, send an email, or view a map. Each activity is given a window in which to draw its UI. The window typically fills the screen, but may be smaller than the screen and float on top of other windows.
+ 
+ Each activity can then start another activity in order to perform different actions. You generally have the main activity which launches other activities (or, your application could just be a single activity). Each time a new activity starts, the previous activity is stopped, but the system preserves the activity in a stack (the "back stack"). When a new activity starts, it is pushed onto the back stack and takes user focus.
+ 
+ Note that an activity can hold multiple "fragments," which is very useful. The activity handles passing information between fragments and lets you switch between different fragments.
+
+#### [Fragments](https://developer.android.com/guide/components/fragments.html) (ex. MainActivityFragment.java)
+Fragments are basically what we described as "screens" in our Snapchat example. Each fragment does a separate task and shows different components to the user. If you were making Snapchat, you might have a fragment for each swipeable screen. You could even [nest fragments within fragments](https://developer.android.com/about/versions/android-4.2.html#NestedFragments) if you'd like! Fragments are useful for:
+
+1. Modularity: dividing complex activity code across fragments for better organization and maintenance.
+2. Reusability: placing behavior or UI parts into fragments that can be shared across multiple activities.
+3. Adaptability: representing sections of a UI as different fragments and utilizing different layouts depending on screen orientation and size.
+
+![alt text][fragment]
+
+However, with the Snapchat example, you could also have each swipeable screen be another activity. Now, this sounds confusing. Why have single fragments that you switch through in one activity instead of just having multiple activities? Well, the biggest reason is for when you're designing an app to be used on both phones and tablets.
```

Comment:
Cool! Definitely link this in the Readme, and also talk over the pros/cons
in class?

On Sun, Jun 19, 2016 at 5:34 PM Nora Mohamed <notifications@github.com>
wrote:

> In lesson2/README.md
> <https://github.com/MobileProto16/Mobile-Proto-16/pull/5#discussion_r67624203>
> :
>
> > + An activity provides a screen that users can interact with in order to do something, such as dial the phone, take a photo, send an email, or view a map. Each activity is given a window in which to draw its UI. The window typically fills the screen, but may be smaller than the screen and float on top of other windows.
> > +
> > + Each activity can then start another activity in order to perform different actions. You generally have the main activity which launches other activities (or, your application could just be a single activity). Each time a new activity starts, the previous activity is stopped, but the system preserves the activity in a stack (the "back stack"). When a new activity starts, it is pushed onto the back stack and takes user focus.
> > +
> > + Note that an activity can hold multiple "fragments," which is very useful. The activity handles passing information between fragments and lets you switch between different fragments.
> > +
> > +#### [Fragments](https://developer.android.com/guide/components/fragments.html) (ex. MainActivityFragment.java)
> > +Fragments are basically what we described as "screens" in our Snapchat example. Each fragment does a separate task and shows different components to the user. If you were making Snapchat, you might have a fragment for each swipeable screen. You could even [nest fragments within fragments](https://developer.android.com/about/versions/android-4.2.html#NestedFragments) if you'd like! Fragments are useful for:
> > +
> > +1. Modularity: dividing complex activity code across fragments for better organization and maintenance.
> > +2. Reusability: placing behavior or UI parts into fragments that can be shared across multiple activities.
> > +3. Adaptability: representing sections of a UI as different fragments and utilizing different layouts depending on screen orientation and size.
> > +
> > +![alt text][fragment]
> > +
> > +However, with the Snapchat example, you could also have each swipeable screen be another activity. Now, this sounds confusing. Why have single fragments that you switch through in one activity instead of just having multiple activities? Well, the biggest reason is for when you're designing an app to be used on both phones and tablets.
>
> in the image, it shows how developing for both phones and tablets could
> lead you to different choices. so, two fragments in one activity versus two
> activities.
> one activity and multiple fragments isn't necessarily the advised way to
> do it. from what i've seen there isn't a standard and it's based on
> opinion/what your app does
>
>
> https://stackoverflow.com/questions/12154505/one-activity-and-all-other-fragments
>
> —
> You are receiving this because you commented.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/MobileProto16/Mobile-Proto-16/pull/5/files/2a171b93dbd9894dde091585847a1fb530a9feb3#r67624203>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AIhRRSISot1AhxaCwTG9e0fLM3EBQjmsks5qNbXhgaJpZM4I5Gov>
> .
>